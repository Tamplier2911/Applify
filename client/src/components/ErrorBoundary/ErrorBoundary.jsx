// import "./ErrorBoundary.scss";
import React, { Component } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { openModal } from "../../redux/modal/modal.actions";

// JS rendering CSS
import {
  ErrorBoundaryContainer,
  ErrorBoundaryErrorSVG,
  ErrorBoundaryText,
  ErrorBoundaryTitle,
  ErrorBoundaryCry
} from "./ErrorBoundaryStyles";

// component constants
import errorBoundaryData from "./ErrorBoundaryConstants";

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hasErrored: false
    };
  }

  static getDerivedStateFromError(error) {
    return { hasErrored: true };
  }

  componentDidCatch(error, info) {
    const { openModal } = this.props;
    console.log(error.message, "error");
    openModal({
      header: "Error!",
      content: "You likely experienced connection issues!"
    });
  }

  render() {
    const { hasErrored } = this.state;
    const { lang } = this.props;
    const { errorBoundaryTitle, errorBoundaryCry } = errorBoundaryData[lang];
    if (hasErrored) {
      return (
        <ErrorBoundaryContainer>
          <ErrorBoundaryErrorSVG />
          <ErrorBoundaryText>
            <ErrorBoundaryTitle>{errorBoundaryTitle}</ErrorBoundaryTitle>
            <ErrorBoundaryCry>{errorBoundaryCry}</ErrorBoundaryCry>
          </ErrorBoundaryText>
        </ErrorBoundaryContainer>
      );
    } else {
      return this.props.children;
    }
  }
}

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps, { openModal })(ErrorBoundary);
