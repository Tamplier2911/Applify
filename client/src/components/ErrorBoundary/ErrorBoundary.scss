/*

.error-boundary {
  display: grid;
  grid-template-rows: min-content 1fr;
  background-color: var(--cl-contacts-bg);
  box-shadow: 0rem 0.1rem 0.4rem var(--cl-square-shadow);
  padding: 2rem;
  margin-bottom: 5rem;

  &__text {
    display: grid;
    justify-items: center;
    text-align: center;
    font-family: var(--font-title);
    font-weight: 300;
  }

  &__title {
    line-height: 1;
    font-size: 5rem;
    align-self: end;
  }

  &__cry {
    align-self: start;
    font-size: 3rem;
  }
}

*/
