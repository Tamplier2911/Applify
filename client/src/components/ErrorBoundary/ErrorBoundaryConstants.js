const errorBoundaryData = {
  eng: {
    errorBoundaryTitle: "How unfortunate!",
    errorBoundaryCry: "ಥ_ಥ"
  },
  rus: {
    errorBoundaryTitle: "Нам очень жаль!",
    errorBoundaryCry: "ಥ_ಥ"
  },
  ukr: {
    errorBoundaryTitle: "Нам дуже прикро!",
    errorBoundaryCry: "ಥ_ಥ"
  }
};

export default errorBoundaryData;
