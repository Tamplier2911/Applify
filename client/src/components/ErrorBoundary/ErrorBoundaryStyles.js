import styled from "styled-components";

// svg
import { ReactComponent as ErrorSVG } from "../../assets/svg/error.svg";

export const ErrorBoundaryContainer = styled.div`
  display: grid;
  grid-template-rows: min-content 1fr;
  background-color: var(--cl-contacts-bg);
  box-shadow: 0rem 0.1rem 0.4rem var(--cl-square-shadow);
  padding: 2rem;
  margin-bottom: 5rem;
`;

export const ErrorBoundaryErrorSVG = styled(ErrorSVG)`
  width: 100%;
`;

export const ErrorBoundaryText = styled.div`
  display: grid;
  justify-items: center;
  text-align: center;
  font-family: var(--font-title);
  font-weight: 300;
`;

export const ErrorBoundaryTitle = styled.div`
  line-height: 1;
  font-size: 5rem;
  align-self: end;
`;

export const ErrorBoundaryCry = styled.div`
  align-self: start;
  font-size: 3rem;
`;
