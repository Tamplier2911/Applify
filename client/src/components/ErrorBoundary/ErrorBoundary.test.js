import React from "react";
import toJson from "enzyme-to-json";

// component
import ErrorBoundary from "./ErrorBoundary";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<ErrorBoundary />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<ErrorBoundary store={store} />);
    wrapper.setState({ hasErrored: true });
  });

  it("renders without an error", () => {
    const errorBoundary = findByTestAttr(wrapper, "error-boundary");
    expect(errorBoundary.length).toBe(1);
  });

  it("renders boundary content", () => {
    const errorBoundaryContent = findByTestAttr(wrapper, "error-boundary-text");
    expect(errorBoundaryContent.length).toBe(1);
  });

  it("matches snapshot", () => {
    const errorBoundary = findByTestAttr(wrapper, "error-boundary");
    expect(toJson(errorBoundary)).toMatchSnapshot();
  });
});
