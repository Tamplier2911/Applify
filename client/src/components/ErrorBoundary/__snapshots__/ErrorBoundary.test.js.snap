// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> matches snapshot 1`] = `
<styled.div
  data-test="error-boundary"
>
  <Styled(Component) />
  <styled.div
    data-test="error-boundary-text"
  >
    <styled.div>
      How unfortunate!
    </styled.div>
    <styled.div>
      ಥ_ಥ
    </styled.div>
  </styled.div>
</styled.div>
`;
